src/main/java/com/google/sps/servlets/study-resources/YouTubeServlet.java

```diff
@@ -68,43 +78,92 @@ public static YouTube getService() throws GeneralSecurityException, IOException
     }
 
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {     
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      System.out.println("GET Request: YouTubeServlet");     
       try {
         YouTube youtubeService = getService();
-        long results = 4;
+        long results = NUM_VIDEOS;
         // Use for creating links
         String embedTemplateLink = "https://www.youtube.com/embed/";
         String directTemplateLink = "https://www.youtube.com/watch?v=";
         String embedUrl = "";
         String directUrl = "";
+        String previousSearchTerm = "";
+        String currentSearchTerm = (String) request.getSession(false).getAttribute("searchKeyword");
         ResourceId videoId = new ResourceId();
-        /* IF NEEDED: Use for video title
         SearchResultSnippet videoInfoSnippet = new SearchResultSnippet();
-        String videoTitle = "";*/
-            
+        String videoTitle = "";
+
+        // Check if the search term has programming in it (to filter out unrelated things)
+        currentSearchTerm = currentSearchTerm.toLowerCase();
```

Comment:
Since indexOf is not really used as an index, can be represented as a boolean "hasProgrammingTerm" instead for clarity.

Also, in-built Java function contains can be used here, something like:
if (!currentSearchTerm.toLowerCase().split("").contains("programming")):
  currentSearchTerm = currentSearchTerm + " programming";

Reply:
I had it originally like that, but split it up because I was getting some issues. I'll probably go back to that format. I'll change it to a Boolean, that makes more sense